### ISSUE TITLE
Fluctuant lr?

### ISSUE THREAD

**Issue by imzhangmf** [NVIDIA/flownet2-pytorch#170]
I wanted to train the model using command as follow:
`$ python main.py --batch_size 8 --model FlowNet2C --optimizer=Adam --optimizer_lr=1e-4 --loss=MultiScale --loss_norm=L1 \                     
--loss_numScales=5 --loss_startScale=4 --optimizer_lr=1e-4 --crop_size 384 512 \
--training_dataset FlyingChairs --training_dataset_root ../data/FlyingChairs/FlyingChairs_release/data  \
--validation_dataset MpiSintelClean --validation_dataset_root ../data/MPI-Sintel-complete/training`

However, my lr is changing every batch, while the output "load" was the lr I set.
The output looked like this: 
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.235, lr: 0.367, load: 1.0e-04:   1%|   | 16/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.235, lr: 0.367, load: 1.0e-04:   1%| | 17/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.242, lr: 0.379, load: 1.0e-04:   1%| | 17/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.242, lr: 0.379, load: 1.0e-04:   1%|   | 18/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.246, lr: 0.391, load: 1.0e-04:   1%|   | 18/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.246, lr: 0.391, load: 1.0e-04:   1%|   | 19/2859.0`
`Training Epoch 1 ['MultiScale-L1', 'EPE']: 0.235, lr: 0.374, load: 1.0e-04:   1%|   | 19/2859.0`

I have just clone the latest repo and never change any code, is this supposed to happen?

**Comment by imzhangmf** [comment]
I think I solved it.

I found that the format of my output is not correct:
`['MultiScale-L1', 'EPE']: 0.242, lr: 0.379, load: 1.0e-04`
which should be:
`MultiScale-L1: 0.242, EPE: 0.379, lr: 1.0e-04, load: ...`
So I go through the code, 
and change the original code:
```
# gather loss_labels, direct return leads to recursion limit error as it looks for variables to gather'
loss_labels = list(model.module.loss.loss_labels)
```
to:
```
# gather loss_labels, direct return leads to recursion limit error as it looks for variables to gather'
if type(model.module.loss.loss_labels) is tuple:
    loss_labels = list(model.module.loss.loss_labels[0])
else:
    loss_labels = list(model.module.loss.loss_labels)
```
then it works fine.

I don't think I'm the only one who encountered this problem.🙈

**Comment by duduheihei** [comment]
> I think I solved it, and this might be a small bug? @fitsumreda
> 
> I found that the format of my output is not correct:
> `['MultiScale-L1', 'EPE']: 0.242, lr: 0.379, load: 1.0e-04`
> which should be:
> `MultiScale-L1: 0.242, EPE: 0.379, lr: 1.0e-04, load: ...`
> So I go through the code,
> and change the original code `main.py`:
> 
> ```
> # gather loss_labels, direct return leads to recursion limit error as it looks for variables to gather'
> loss_labels = list(model.module.loss.loss_labels)
> ```
> 
> to:
> 
> ```
> # gather loss_labels, direct return leads to recursion limit error as it looks for variables to gather'
> if type(model.module.loss.loss_labels) is tuple:
>     loss_labels = list(model.module.loss.loss_labels[0])
> else:
>     loss_labels = list(model.module.loss.loss_labels)
> ```
> 
> then it works fine.
> 
> I don't think I'm the only one who encountered this problem.🙈

I got the same error and fixed it. Now I try to train flownet2C along but failed, Do you train FlowNet2C  succeed? My qq: 616291904,  hope we can exchange ideas.
